src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java

```diff
@@ -467,4 +467,31 @@ public static String getFileExtension(String fileNameWithExtension) {
         }
         return extension;
     }
+
+    /**
+     * Checks whether the given string is a valid identifier.
+     * @param str A string to check.
+     * @return true when the given string contains valid identifier.
+     */
+    public static boolean isIdentifier(String str) {
+        boolean isIdentifier = !str.isEmpty() && Character.isJavaIdentifierStart(str.charAt(0));
```

Comment:
@soon See my other comment.
>  charAt(0) part of the for loop

`if` statement should be `if (!str.isEmpty()) {` alone.
`for` loop should be `for (int i = 0; i < str.length(); i++) {`.